client/ayon_deadline/plugins/publish/global/collect_jobinfo.py

```diff
@@ -0,0 +1,237 @@
+# -*- coding: utf-8 -*-
+import pyblish.api
+from ayon_core.lib import (
+    BoolDef,
+    NumberDef,
+    TextDef,
+    EnumDef,
+    is_in_tests,
+    UISeparatorDef
+)
+from ayon_core.pipeline.publish import AYONPyblishPluginMixin
+from ayon_core.lib.profiles_filtering import filter_profiles
+
+from ayon_deadline.lib import FARM_FAMILIES, AYONDeadlineJobInfo
+
+
+class CollectJobInfo(pyblish.api.InstancePlugin, AYONPyblishPluginMixin):
+    """Collect variables that belong to Deadline's JobInfo.
+
+    Variables like:
+    - department
+    - priority
+    - chunk size
+
+    """
+
+    order = pyblish.api.CollectorOrder + 0.420
+    label = "Collect Deadline JobInfo"
+
+    families = FARM_FAMILIES
+    targets = ["local"]
+
+    def process(self, instance):
+        attr_values = self._get_jobinfo_defaults(instance)
+
+        attr_values.update(self.get_attr_values_from_data(instance.data))
+        # do not set empty strings
+        attr_values = {
+            key: value
+            for key,value in attr_values.items()
+            if value != ""
+        }
+        job_info = AYONDeadlineJobInfo.from_dict(attr_values)
+        instance.data["deadline"]["job_info"] = job_info
+
+    @classmethod
+    def get_attr_defs_for_instance(cls, create_context, instance):
+        cls.log.info(create_context.get_current_task_entity())
+        if not cls.instance_matches_plugin_families(instance):
+            return []
+
+        # will be reworked when CreateContext contains settings and task types
+        project_name = create_context.project_name
+        project_settings = (
+            create_context.get_current_project_settings()
+        )
+
+        profiles = (
+            project_settings["deadline"]["publish"][cls.__name__]["profiles"])
```

Comment:
Isn't this just as easily globally done via `apply_settings` logic that this needs to be done one instead of per instance?

Reply:
100% agree, just store profiles during `apply_settings`.